src/utils/error-handle.js

```diff
@@ -0,0 +1,6 @@
+var util = require('util');
+
+module.exports = function handleSystemError(message, ...objs){
```

Comment:
Encapsulating the throwing of an error as a side-effect of this function makes it a lot harder to follow the execution flow when looking at the code (whereas an explicit `throw` would make it immediately explicit). It also adds an unwanted frame on the error stack, which will make it look like all errors are thrown from this file (e.g. in Sentry logs).

Couldn't you simply throw a different type of object (e.g. `throw new InvalidVFocusError('Invalid VFocus passed to toggleSelectedNoteCollapseState', focus)`).

I'm also slightly curious why the need for so many defensive assets; why are the wrong objects passed to these methods in the first place?